### DISCUSSION TITLE
External I2C - Bluetooth module required

### DISCUSSION THREAD

**Discussion by WolleKette** [micropython/micropython#11820] [Hardware & Peripherals]
For an application on a STM32 Nucelo board I need Bluetooth to send and receive data. Unfortunately the board has no bluetooth and I can't use another board.

I was thought of using an external Bluetooth module. Does anyone have experience with this in MicroPython? 
Which module and which Bluetooth library did you use?

**Comment by jimmo** [comment]
@WolleKette typically most external Bluetooth/BLE controllers are connected via a UART (via the HCI protocol).

One of the easiest options by far is to use an nRF51 or nRF52 running Zephyr's HCI UART firmware as an external host controller. MicroPython's BLE support (using both NimBLE and BlueKitchen) will work with this. https://docs.zephyrproject.org/latest/samples/bluetooth/hci_uart/README.html
